props/saber_shtok_buttons.h

```diff
@@ -64,37 +177,170 @@ SaberShtokButtons() : PropBase() {}
-#ifdef DUAL_POWER_BUTTONS
-        aux_on_ = true;
-        On();
-#else
+      case EVENTID(BUTTON_NONE, EVENT_SHAKE, MODE_OFF):
+        if (fusor.angle1() >  M_PI / 4) {
         next_preset();
+        }
+        return true;
+
+
+// Activate Muted
+      case EVENTID(BUTTON_POWER, EVENT_DOUBLE_CLICK, MODE_ON):
+        if (millis() - activated_ < 500) {
+          if (SetMute(true)) {
+            unmute_on_deactivation_ = true;
+          }
+        }
+        return true;
+
+// Turn Blade OFF Button
+      case EVENTID(BUTTON_POWER, EVENT_HELD_LONG, MODE_ON):
+        if (!SaberBase::Lockup()) {
+#ifndef DISABLE_COLOR_CHANGE
+          if (SaberBase::GetColorChangeMode() != SaberBase::COLOR_CHANGE_MODE_NONE) {
+            // Just exit color change mode.
+            // Don't turn saber off.
+            ToggleColorChangeMode();
+            return true;
+          }
 #endif
+          Off();
+ 	saber_off_time_ = millis();
+        swing_blast_ = false;
+        battle_mode_ = false;
+      }
+        return true;
+
+// Turn Blade OFF Twist
+      case EVENTID(BUTTON_NONE, EVENT_TWIST, MODE_ON):
+        if (!SaberBase::Lockup()) {
+#ifndef DISABLE_COLOR_CHANGE
+          if (SaberBase::GetColorChangeMode() != SaberBase::COLOR_CHANGE_MODE_NONE) {
+            // Just exit color change mode.
+            // Don't turn saber off.
+            ToggleColorChangeMode();
+            return true;
+          }
+#endif
+        if (fusor.angle1() <  M_PI / 4) {
+          Off();
+ 	saber_off_time_ = millis();
+        swing_blast_ = false;
+        battle_mode_ = false;
+        }
+      }
+        return true;
+
+// Force
+      case EVENTID(BUTTON_POWER, EVENT_CLICK_SHORT, MODE_ON):
+        SaberBase::DoForce();
+        return true;
+
+// Color Change mode
+#ifndef DISABLE_COLOR_CHANGE
+      case EVENTID(BUTTON_NONE, EVENT_SHAKE, MODE_ON):
+        if (fusor.angle1() >  M_PI / 4) {
+        ToggleColorChangeMode();
+        }
+        break;
+#endif
+
+// Blaster Deflection
+      case EVENTID(BUTTON_AUX, EVENT_CLICK_SHORT, MODE_ON):
+        swing_blast_ = false;
+        SaberBase::DoBlast();
+        return true;
+
+// Multi-Blaster Deflection mode
+
+// Push
+      case EVENTID(BUTTON_NONE, EVENT_PUSH, MODE_ON):
+        if (millis() - last_push_ > 2000) {
+          if (SFX_push) {
+            hybrid_font.PlayCommon(&SFX_push);
+          } else {
+            hybrid_font.DoEffect(EFFECT_FORCE, 0);
+          }
+          last_push_ = millis();
+        }
+        return true;
+
+// Lockup
+      case EVENTID(BUTTON_AUX, EVENT_HELD, MODE_ON):
+        if (!SaberBase::Lockup()) {
+          if (pointing_down_) {
+            SaberBase::SetLockup(SaberBase::LOCKUP_DRAG);
+          } else {
+            SaberBase::SetLockup(SaberBase::LOCKUP_NORMAL);
+          }
+          swing_blast_ = false;
+          SaberBase::DoBeginLockup();
+          return true;
+        }
+        break;
+
+// Lightning Block
```

Comment:
This comment seems like it's not for anything.
Maybe remove it?

Reply:
Deleted.